utils/netutils/netutils.go

```diff
@@ -181,6 +185,156 @@ func GetIPNumber(subnetIP string, subnetLen uint, allocSubnetLen uint, hostIP st
 	return uint(hostID), nil
 }
 
+// ReserveIPv6HostID sets the hostId in the AllocMap
+func ReserveIPv6HostID(hostID string, IPv6AllocMap *map[string]bool) {
+	if hostID == "" {
+		return
+	}
+	if *IPv6AllocMap == nil {
+		*IPv6AllocMap = make(map[string]bool)
+	}
+	(*IPv6AllocMap)[hostID] = true
+}
+
+// GetNextIPv6HostID returns the next available hostId in the AllocMap
+func GetNextIPv6HostID(hostID, subnetAddr string, subnetLen, allocSubnetLen uint, IPv6AllocMap map[string]bool) (string, error) {
```

Comment:
IPv4 address allocator takes separate subnetLen and allocSubnetLen for some historical reason. Which is not relevant anymore. You can probably simplify the logic here if you remove allocSubnetLen variable and assume 128bit IPv6 addresses..